x11-wm/mutter/mutter-43.0.ebuild

```diff
@@ -0,0 +1,200 @@
+#	X? (
+DEPEND+="
+		>=x11-libs/libXcomposite-0.4
+		x11-libs/libXcursor
+		x11-libs/libXdamage
+		x11-libs/libXext
+		>=x11-libs/libXfixes-3
+		>=x11-libs/libXi-1.7.4
+		x11-libs/libXtst
+		x11-libs/libxkbfile
+		x11-misc/xkeyboard-config
+		>=x11-libs/libxkbcommon-0.4.3[X]
+		x11-libs/libXrender
+		>=x11-libs/libXrandr-1.5.0
+		x11-libs/libxcb:=
+		x11-libs/libXinerama
+		x11-libs/libXau
+"
+#	)"
+
+RDEPEND="${DEPEND}
+	gnome-extra/zenity
+
+	!<gui-libs/gtk-4.6.4:4
+	!<x11-libs/gtk+-3.24.34:3
+"
+DEPEND="${DEPEND}
+	x11-base/xorg-proto
+	sysprof? ( >=dev-util/sysprof-common-3.38.0 )
+"
+BDEPEND="
+	dev-util/wayland-scanner
+	dev-util/gdbus-codegen
+	dev-util/glib-utils
+	>=sys-devel/gettext-0.19.8
+	virtual/pkgconfig
+	test? (
+		${PYTHON_DEPS}
+		$(python_gen_any_dep '
+			dev-python/python-dbusmock[${PYTHON_USEDEP}]
+		')
+		app-text/docbook-xml-dtd:4.5
+		x11-misc/xvfb-run
+	)
+	wayland? (
+		>=sys-kernel/linux-headers-4.4
+		x11-libs/libxcvt
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-43.0-Disable-anonymous-file-test.patch
+)
+
+python_check_deps() {
+	if use test; then
+		python_has_version "dev-python/python-dbusmock[${PYTHON_USEDEP}]"
+	fi
+}
+
+src_configure() {
+	# TODO: Replicate debug vs release meson build type behaviour under our buildtype=plain
+	local emesonargs=(
+		-Dopengl=true
+		#opengl_libname
+		#gles2_libname
+		$(meson_use wayland gles2)
+		-Degl=true
+		-Dglx=true
+		$(meson_use wayland)
+		$(meson_use systemd)
+		$(meson_use wayland native_backend)
+		$(meson_use screencast remote_desktop)
+		-Dlibgnome_desktop=true
+		$(meson_use udev)
+		-Dudev_dir=$(get_udevdir)
+		$(meson_use input_devices_wacom libwacom)
+		$(meson_use sound sound_player)
+		-Dpango_ft2=true
+		-Dstartup_notification=true
+		-Dsm=true
+		$(meson_use introspection)
+		$(meson_use doc docs)
+		$(meson_use test cogl_tests)
+		$(meson_use wayland core_tests) # core tests require wayland; overall -Dtests option is honored on top, so no extra conditional needed
+		-Dnative_tests=false
+		$(meson_use test clutter_tests)
+		$(meson_use test tests)
+		-Dkvm_tests=false
+		-Dtty_tests=false
+		$(meson_use sysprof profiler)
+		-Dinstalled_tests=false
+
+		#verbose # Let upstream choose default for verbose mode
+		#xwayland_path
+		# TODO: relies on default settings, but in Gentoo we might have some more packages we want to give Xgrab access (mostly virtual managers and remote desktops)
+		#xwayland_grab_default_access_rules
+	)
+
+	if use wayland && use video_cards_nvidia; then
+		emesonargs+=(
+			-Degl_device=true
+			-Dwayland_eglstream=true
+		)
+	else
+		emesonargs+=(
+			-Degl_device=false
+			-Dwayland_eglstream=false
+		)
+	fi
+
+	meson_src_configure
+}
+
+src_test() {
+	gnome2_environment_reset # Avoid dconf that looks at XDG_DATA_DIRS, which can sandbox fail if flatpak is installed
+	glib-compile-schemas "${BUILD_DIR}"/data
+	GSETTINGS_SCHEMA_DIR="${BUILD_DIR}"/data meson_src_test --setup=CI
```

Comment:
Do the tests work for you? They timeout and fail for me:

````
  1/195 mutter:cogl+cogl/conform / cogl-test-atlas-migration-gl                                                 TIMEOUT        300.00s   (exit status 133 or signal 5 SIGTRAP)
>>> MUTTER_DEBUG_DUMMY_MODE_SPECS=800x600@10.0 G_TEST_SRCDIR=/var/tmp/portage/x11-wm/mutter-43.0/work/mutter-43.0/src/tests/cogl/conform G_TEST_BUILDDIR=/var/tmp/portage/x11-wm/mutter-43.0/work/mutter-43.0-build/src/tests/cogl/conform MALLOC_PERTURB_=189 COGL_DRIVER=gl MUTTER_TEST_PLUGIN_PATH=/var/tmp/portage/x11-wm/mutter-43.0/work/mutter-43.0-build/src/compositor/plugins/libdefault.so G_ENABLE_DIAGNOSTIC=0 /var/tmp/portage/x11-wm/mutter-43.0/work/mutter-43.0/src/tests/meta-dbus-runner.py -- /usr/bin/xvfb-run -a -s '+iglx -noreset' /var/tmp/portage/x11-wm/mutter-43.0/work/mutter-43.0-build/src/tests/cogl/conform/cogl-test-atlas-migration
 ✀  
stdout:
# random seed: R02S7f496815ebc0e91b12aad76b9ff45808
# libmutter-MESSAGE: Running Mutter Test (using mutter 43.0) as a Wayland display server
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation memory (GMemorySettingsBackend) for ‘gsettings-backend’
# GLib-GIO-DEBUG: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
# libmutter-MESSAGE: Created surfaceless renderer without GPU
# GVFS-DEBUG: org.gtk.vfs.MountTracker.listMountableInfo call failed: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gtk.vfs.Daemon was not provided by any .service files (g-dbus-error-quark, 2)
# 
# GLib-GIO-DEBUG: _g_io_module_get_default: Found default implementation local (GLocalVfs) for ‘gio-vfs’
# libmutter-MESSAGE: Disabling DMA buffer screen sharing (not hardware accelerated)
# libmutter-MESSAGE: Disabling DMA buffer screen sharing (implicit modifiers not supported)
# libmutter-MESSAGE: Using Wayland display name 'mutter-test-display'
1..1
ok 1 /atlas-migration
Bail out! GLib-GObject-FATAL-CRITICAL: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
stderr:
Starting D-Bus daemons (session & system)...
Starting mocked services...
Running test case...
libmutter-Message: 22:28:15.056: Running Mutter Test (using mutter 43.0) as a Wayland display server
libEGL warning: failed to open /dev/dri/renderD128: Permission denied

libEGL warning: failed to open /dev/dri/card0: Permission denied

libmutter-Message: 22:28:15.088: Created surfaceless renderer without GPU
libmutter-Message: 22:28:15.103: Disabling DMA buffer screen sharing (not hardware accelerated)
libmutter-Message: 22:28:15.103: Disabling DMA buffer screen sharing (implicit modifiers not supported)
libmutter-Message: 22:28:15.103: Using Wayland display name 'mutter-test-display'
Window manager warning: Failed to set environment variable WAYLAND_DISPLAY for gnome-session: GDBus.Error:org.freedesktop.DBus.Error.NameHasNoOwner: Name "org.gnome.SessionManager" does not exist
````

Reply:
yes I asked in the irc channel about this, too. Is this because gnome-shell-43.0 is missing? 

This is also the reason I kept this PR draft

Reply:
Nope, have gnome-shell-43.0 installed, still can not get the tests to pass

Reply:
I also tried to run it yesterday, once I got gnome-shell-43.0 installed, and it crashes right from the start, when gdm is started

Reply:
Here is the syslog when I try to start gdm. https://pastebin.com/bWNcPvkM

Reply:
I've bisected two things causing the tests to fail:

- Tests fail with `-Dsound_player=false`: https://gitlab.gnome.org/GNOME/mutter/-/issues/2451
- Tests timeout with python-dbusmock < 0.28: https://gitlab.gnome.org/GNOME/mutter/-/issues/2452

In the interest of getting this into Gentoo so people can actually test it, I suggest that we remove `IUSE=sound` and leave libcanberra support enabled always and that we require python-dbusmock >= 0.28.

Reply:
There is still the problem that it fails to run for me on my openrc machine and works on a systemd laptop I have.
42.3 works fine on both.
Can we get someone else with a openrc machibe to test this so we don't break people's computers accidentally?